spec/bolt_server/transport_app_spec.rb

```diff
@@ -787,82 +788,184 @@ def post_over_transport(transport, action, body_content, multiple: false)
     end
 
     describe '/project_file_metadatas/:module_name/:file' do
-      let(:fake_pal) { instance_double('Bolt::PAL') }
-      let(:fake_project) { instance_double('Bolt::Project') }
-      let(:fake_config) { instance_double('Bolt::Config') }
-      let(:fake_environment) { instance_double('Puppet::Node::Environment') }
-      let(:fake_module) { instance_double('Puppet::Module') }
-      let(:fake_file) { 'foo_file_abs_path' }
-      let(:fake_fileset) { instance_double('Puppet::FileServing::Fileset') }
-      let(:project_ref) { 'some_project_somesha' }
-      let(:module_name) { 'foo_module' }
+      let(:project_ref) { 'bolt_server_test_project' }
+      let(:module_name) { 'project_module' }
       let(:file) { 'foo_file' }
-      let(:path) { "/project_file_metadatas/#{module_name}/#{file}?project_ref=#{project_ref}" }
-
-      before(:each) do
-        allow(Dir).to receive(:exist?).with("/tmp/foo/#{project_ref}").and_return(true)
-        allow(Bolt::Project).to receive(:create_project).and_return(fake_project)
-        allow(Bolt::Config).to receive(:from_project).and_return(fake_config)
-        allow(fake_config).to receive(:modulepath)
-        allow(fake_config).to receive(:project).and_return(fake_project)
-        allow(Bolt::PAL).to receive(:new).and_return(fake_pal)
-        allow(fake_pal).to receive(:in_bolt_compiler).and_yield
-        allow(Puppet).to receive(:lookup).with(:current_environment).and_return(fake_environment)
-        allow(fake_environment).to receive(:module).with(module_name).and_return(fake_module)
-        allow(fake_module).to receive(:file).with(file).and_return(fake_file)
-        # The Puppet::FileServing code will be tested more thoroughly in Orch's acceptance
-        # tests so it is enough for the unit tests to make sure that we're returning a 200
-        # status when the metadata's retrieved.
-        allow(Puppet::FileServing::Fileset).to receive(:new).with(fake_file, anything).and_return(fake_fileset)
-        allow(Puppet::FileServing::Fileset).to receive(:merge).with(fake_fileset).and_return([])
-      end
+      let(:path) { "" }
 
       it 'returns 400 if project_ref is not specified' do
-        path = '/project_file_metadatas/foo_module/foo_file'
-        get(path)
+        get('/project_file_metadatas/foo_module/foo_file')
         error = last_response.body
         expect(error).to eq("`project_ref` is a required argument")
         expect(last_response.status).to eq(400)
       end
 
       it 'returns 400 if project_ref does not exist' do
-        allow(Dir).to receive(:exist?).with("/tmp/foo/#{project_ref}").and_return(false)
-        get(path)
+        get("/project_file_metadatas/bar/foo?project_ref=not_a_real_project")
         error = last_response.body
-        expect(error).to eq("`project_ref`: /tmp/foo/#{project_ref} does not exist")
+        expect(error).to include("`project_ref`: #{project_dir}/not_a_real_project does not exist")
```

Comment:
Yeah, propagating bad consistency is probably not a good idea (e.g. https://github.com/puppetlabs/bolt/blob/main/spec/bolt_server/transport_app_spec.rb#L236 and other places also use `eq`).

I think we use `include`, then the match should be shorter? e.g. `include(%r{project_ref.*#{project_dir}/not_a_real_project)` ?